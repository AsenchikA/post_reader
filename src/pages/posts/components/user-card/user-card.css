.user-card {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 15px;
  border-radius: 5px;
  border: 1px solid blue;
  cursor: pointer;

  &__post-count {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 25px;
    min-width: 25px;
    padding: 0 5px;
    border-radius: 15px;
    background-color: #bebeff;
    color: white;
  }
}
