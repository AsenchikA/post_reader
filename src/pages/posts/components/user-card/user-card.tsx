import React, { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';
import { IUserModel } from 'src/types';
import styles from './user-card.css';

interface IUserCardProps extends IUserModel {
  postCount: number;
  className?: string;
}

export const UserCard: FunctionComponent<IUserCardProps> = ({ id, name, postCount, className }) => {
  return (
    <Link className={className} to={`/${id}`}>
      <div className={styles['user-card']} key={id}>
        <span>{name}</span>
        <div className={styles['user-card__post-count']}>{postCount}</div>
      </div>
    </Link>
  );
};
