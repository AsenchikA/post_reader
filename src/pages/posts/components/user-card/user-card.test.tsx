import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { UserCard } from './user-card';

it('UserCard renders correcly', () => {
  const props = {
    id: '1',
    name: 'user',
    postCount: 1,
    className: 'test-class',
  };

  const tree = renderer
    .create(
      <MemoryRouter>
        <UserCard {...props} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
