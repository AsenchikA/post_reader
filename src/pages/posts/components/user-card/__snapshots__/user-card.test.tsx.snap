// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard renders correcly 1`] = `
<a
  className="test-class"
  href="/1"
  onClick={[Function]}
>
  <div
    className="user-card"
  >
    <span>
      user
    </span>
    <div
      className="user-card__post-count"
    >
      1
    </div>
  </div>
</a>
`;
